Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -661,5 +664,27 @@ public enum ProductUpdateError: Error, Equatable {
 }
 
 public enum ProductLoadError: Error, Equatable {
+    case notFound
     case notFoundInStorage
+    case unknown(error: AnyError)
+
+    init(underlyingError error: Error) {
+        guard case let DotcomError.unknown(code, _) = error else {
+            self = .unknown(error: error.toAnyError)
+            return
+        }
+
+        self = ErrorCode(rawValue: code)?.error ?? .unknown(error: error.toAnyError)
+    }
+
+    enum ErrorCode: String {
+        case invalidId = "woocommerce_rest_product_invalid_id"
```

Comment:
super nit: we generally say `ID` in the codebase 😄 

```suggestion
        case invalidID = "woocommerce_rest_product_invalid_id"
```